operations/patch_util.go

```diff
@@ -306,6 +304,11 @@ func (p *patchParams) setNonRepeatedDefaults(conf *ClientSettings) {
 	}
 }
 
+// tasksAndVariantsSet is true if both tasks and variants were specified directly or via regex.
+func (p *patchParams) tasksAndVariantsSet() bool {
```

Comment:
nit: it's slightly more idiomatic to call a method that returns a bool something like `hasTasksAndVariants`.

Reply:
ah yeah i was going to do that but then i bailed because it felt unclear, but you're right it should be fine especially for a private function